CheckerboardKarel.java

```diff
@@ -1,16 +1,108 @@
-/*
- * File: CheckerboardKarel.java
- * ----------------------------
- * When you finish writing it, the CheckerboardKarel class should draw
- * a checkerboard using beepers, as described in Assignment 1.  You
- * should make sure that your program works for all of the sample
- * worlds supplied in the starter folder.
- */
-
-import stanford.karel.*;
-
-public class CheckerboardKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: CheckerboardKarel.java
+ * ----------------------------
+ * When you finish writing it, the CheckerboardKarel class should draw
+ * a checkerboard using beepers, as described in Assignment 1.  You
+ * should make sure that your program works for all of the sample
+ * worlds supplied in the starter folder.
+ */
+
+import stanford.karel.*;
+
+public class CheckerboardKarel extends SuperKarel {

```

Comment:
Your solution assumes that you are dealing with the world that is 8x8 but the assignment said you can have worlds with different dimensions. Because of the assumption you made your code doesn't work when the world size is not 8x8. Also you need to do a better job at breaking down your solution into methods with fewer lines of code (The run method is too long).